site/zh-cn/beta/tutorials/keras/basic_text_classification_with_tfhub.ipynb

```diff
@@ -0,0 +1,471 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "_NUbzVeYkgcO"
+   },
+   "outputs": [],
+   "source": [
+    "embedding = \"https://tfhub.dev/google/tf2-preview/gnews-swivel-20dim/1\"\n",
+    "hub_layer = hub.KerasLayer(embedding, input_shape=[], \n",
+    "                           dtype=tf.string, trainable=True)\n",
+    "hub_layer(train_examples_batch[:3])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "dfSbV6igl1EH"
+   },
+   "source": [
+    "现在让我们构建完整模型："
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "xpKOoWgu-llD"
+   },
+   "outputs": [],
+   "source": [
+    "model = tf.keras.Sequential()\n",
+    "model.add(hub_layer)\n",
+    "model.add(tf.keras.layers.Dense(16, activation='relu'))\n",
+    "model.add(tf.keras.layers.Dense(1, activation='sigmoid'))\n",
+    "\n",
+    "model.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "6PbKQ6mucuKL"
+   },
+   "source": [
+    "层按顺序堆叠以构建分类器：\n",
+    "\n",
+    "1. 第一层是 Tensorflow Hub 层。这一层使用一个预训练的保存好的模型来将句子映射为嵌入向量（embedding vector）。我们所使用的预训练文本嵌入（embedding）模型([google/tf2-preview/gnews-swivel-20dim/1](https://tfhub.dev/google/tf2-preview/gnews-swivel-20dim/1))将句子切割为符号，嵌入（embed）每个符号然后进行合并。最终得到的维度是：`(num_examples, embedding_dimension)`。\n",
+    "2. 该定长输出向量通过一个有 16 个隐层单元的全连接层（`Dense`）进行管道传输。\n",
+    "3. 最后一层与单个输出结点紧密相连。使用 `Sigmoid` 激活函数，其函数值为介于 0 与 1 之间的浮点数，表示概率或置信水平。\n",
+    "\n",
+    "让我们编译模型。"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "L4EqVWg4-llM"
+   },
+   "source": [
+    "### 损失函数与优化器\n",
+    "\n",
+    "一个模型需要损失函数和优化器来进行训练。由于这是一个二分类问题且模型输出概率值（一个使用 sigmoid 激活函数的单一单元层），我们将使用 `binary_crossentropy` 损失函数。\n",
+    "\n",
+    "这不是损失函数的唯一选择，例如，您可以选择 `mean_squared_error` 。但是，一般来说 `binary_crossentropy` 更适合处理概率——它能够度量概率分布之间的“距离”，或者在我们的示例中，指的是度量 ground-truth 分布与预测值之间的“距离”。\n",
+    "\n",
+    "稍后，当我们研究回归问题（例如，预测房价）时，我们将介绍如何使用另一种叫做均方误差的损失函数。\n",
+    "\n",
+    "现在，配置模型来使用优化器和损失函数："
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "Mr0GP-cQ-llN"
+   },
+   "outputs": [],
+   "source": [
+    "model.compile(optimizer='adam',\n",
+    "              loss='binary_crossentropy',\n",
+    "              metrics=['accuracy'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "35jv_fzP-llU"
+   },
+   "source": [
+    "## 训练模型\n",
+    "\n",
+    "在 mini-batche 大小为 512 个样本的条件下迭代 20 个 epoch 来训练模型。 这是指对 `x_train` 和 `y_train` 张量中所有样本的的 20 次迭代。在训练过程中，监测来自验证集的 10,000 个样本上的损失值（loss）和准确率（accuracy）："
```

Comment:
原文
Train the model for 20 epochs in mini-batches of 512 samples

建议翻译为
训练一个小批量的512个样本模型迭代20次

Reply:
Thanks for reviewing.  I've made some modifications to your opinion except this one.  Since `mini-batch` and `epoch` are proper nones in machine learning, I think they should be retained and not translated. And `mini-batches of 512 samples` means every training step has 512 samples but not train the model with just 512 samples, for this reason, I hope to retain the previous translation. We can discuss further to determine the final translation result.

Reply:
Or `以 512 个样本的 mini-batch 大小迭代 20 个 epoch 来训练模型。`. If there are other suggestions, please tell me. And thanks again.

Reply:
nothing , you are right